Update user logic

bolajahmad opened the pull request:
This PR adds logic to allows users to update their registered information. Users are allowed to update information like,

- Email address
- Telegram handle
- Notifiers desired

User ID must be provided

**Fixes**
Adds a fix to prevent users from registering with an ID. IDs must be autogenerated by the DB.
Refactored code in the users db crate.

bolajahmad committed:
update: Initialized a new route to update user information; added new db schema to update user

bolajahmad committed:
update: hooked up updatuser route

bolajahmad committed:
lint: cargo format fix-all formatting errors

bolajahmad committed:
update: hooked up API for update user to store users on DB

Szegoo commented:
> Adds a fix to prevent users from registering with an ID. IDs must be autogenerated by the DB.

I changed this on purpose. The current `u32` we use for id is just a place holder. This will be the address of the user.

bolajahmad committed:
restore logic to register users with an ID, no autogenerate

bolajahmad commented:
> > Adds a fix to prevent users from registering with an ID. IDs must be autogenerated by the DB.
> 
> I changed this on purpose. The current `u32` we use for id is just a place holder. This will be the address of the user. Please revert the change

Yes, as we discussed I already reverted this logic. The ID is the address so it must be provided during registration and cannot be changed once registered.

Szegoo commented on services/api/routes/src/update.rs:
```diff
@@ -1 +1,94 @@
+** Validate user by exists by using only the ID.
+** If the ID exists, then it can be validated.
+**
+** How can we ensure that the owner of the email | tg is making the update?
```
Here are the details how I envisioned the authentication to look like:
1. User first registers an account and sets a notifier. During the registration process the user proves that they are the owner of the id they provided(the id will be a substrate address, and validation will be done by requesting the user to sign some 'welcome message'. The user will also prove that they are the notifier owner(e.g. when using email prove that they are the email owner).

2. After successfully proving, the server will give the user a JWT, which will be stored in the users browser.

3. When updating, the user will also send along the JWT, and after validating it we know that the user made the call and we can update the user data. However, when changing the notifier we need to validate that the user is actually the owner of the new selected notifier. This can be done the same way as in the registration process.

Part of the first step is getting implemented in this PR: #11

Szegoo committed:
update docs formatting

bolajahmad committed:
add unit tests to update api

bolajahmad committed:
fix: organized code imports

bolajahmad committed:
lint: added code formatting using cargo format
